src/datasets/queries/time-series.query.ts

```diff
@@ -0,0 +1,118 @@
+import { XYPoint } from '../metas/types';
+import { inOneOfDateRanges } from '../../models/data-cube/filters';
+import { DAY } from '../../utils/timeUnits';
+import { DataCube } from '../../models/data-cube/data-cube.model';
+import { ResultRow } from '../../models/data-cube/types';
+import { unique } from '../../utils/misc';
+
+export interface TimeSeriesQueryOptions {
+  range: [Date, Date];
+}
+
+export type TimeSeriesPoint = XYPoint<Date, number>;
+
+export interface TimeSeriesDatum<S> {
+  label: string;
+  style?: Partial<S>;
+  points: TimeSeriesPoint[];
+}
+
+export type TimeSeriesQuery<S> = (options: TimeSeriesQueryOptions) => TimeSeriesDatum<S>[];
+
+export type LegendItem<S> = {
+  label: string;
+  style?: Partial<S>;
+  measureName: string;
+  periodOffset?: number;
+  windowSize?: number;
+};
+
+export function createTimeSeriesQuery<S>(dataCube: DataCube, legendItems: LegendItem<S>[]): TimeSeriesQuery<S> {
```

Comment:
Please add a JSDoc for this exported function.

Reply:
@jnager Sure. A related question: what kinds of exported functions need JSDoc? Is it necessary to add JSDoc to all the exported functions?

@64json would you mind me to do that?

Reply:
@jcc37 I'd appreciate if you can work on it!

Reply:
Our internal styleguide requires JSDoc comments for every top-level export of a module and generally recommends comments for any method or properties whose purpose is not obvious from name/type (often judged by reviewers).

I don't think you need to add JSDoc comments on every function since this is external to our codebase, but I personally find comments more helpful than not usually, and I do think they helps reviews go faster!